fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/ExplainTokens.java

```diff
@@ -0,0 +1,1110 @@
+     * to be cut off).
+     * @param renderingExplainLevel the desired {@link ExplainLevel}
+     * @param formatter the formatter
+     * @param remainingCharacterBudget the budget of the number of remaining characters
+     * @return a {@link CharSequence} representing the rendered explain string which is potentially cut off if
+     *         depending on the budget. The returned character sequence is at most of length
+     *         {@code remainingCharacterBudget + 3} to accommodate for the {@code ...}. Note that using
+     *         {@link CharSequence} instead of {@link String} is preferable as {@link StringBuilder} is a
+     *         {@link CharSequence} allowing us to return un-built string builders without materializing them to
+     *         strings.
+     */
+    @Nonnull
+    public CharSequence render(final int renderingExplainLevel,
+                               @Nonnull final ExplainFormatter formatter,
+                               int remainingCharacterBudget) {
+        final StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < tokens.size(); i++) {
+            final var token = tokens.get(i);
+            final var stringedToken =
+                    token.render(renderingExplainLevel, formatter, remainingCharacterBudget);
+            stringBuilder.append(stringedToken);
+            remainingCharacterBudget -= (remainingCharacterBudget == Integer.MAX_VALUE ? 0 : stringedToken.length());
+            if (remainingCharacterBudget < 1) {
+                if (remainingCharacterBudget == 0 && i + 1 < tokens.size()) {
+                    // if any of the remaining tokens render to more than an empty string we have to append ... instead
+                    for (int j = i + 1; j < tokens.size(); j++) {
+                        final var remainingToken = tokens.get(j);
+                        final var remainingStringedToken =
+                                remainingToken.render(renderingExplainLevel, formatter, 0);
+                        if (!remainingStringedToken.toString().isEmpty()) {
+                            return stringBuilder.append(remainingStringedToken);
+                        }
+                    }
+                }
+                return stringBuilder;
+            }
+        }
+        return stringBuilder;
+    }
+
+    @Nonnull
+    public ExplainTokens add(@Nonnull final Token token) {
+        tokens.add(token);
+        this.tokenSizes = vectorAdd(this.tokenSizes, token.getTokenSizes());
+        this.minLengths = vectorAdd(this.minLengths, token.getMinLengths());
+        this.maxLengths = vectorAdd(this.maxLengths, token.getMaxLengths());
+        return this;
+    }
+
+    @Nonnull
+    public ExplainTokens addNested(@Nonnull final ExplainTokens additionalExplainTokens) {
+        addNested(Token.DEFAULT_EXPLAIN_LEVEL, additionalExplainTokens);
+        return this;
```

Comment:
```suggestion
        return addNested(Token.DEFAULT_EXPLAIN_LEVEL, additionalExplainTokens);
```

There seems to be a mix of styles here, where some overloads of `add` are written as:

```java
public ExplainTokens addThing(T thing) {
   addThingWithOtherMethod(thing);
   return this;
}
```

And some are written:

```java
public ExplainTokens addThing(T thing) {
   return addThingWithOtherMethod(thing);
}
```

I think the latter is slightly more idiomatic, so I think we should standardize on that, but it's a bit six of one, half dozen of another.

Reply:
Changed all the ones I could find to the more concise second form.